lib/resty/etcd/v3.lua

```diff
@@ -688,43 +689,77 @@ local function request_chunk(self, method, path, opts, timeout)
 
 
     local function read_watch()
+        body = ""
 
         while(1) do
-            body, err = res.body_reader()
+            local chunk, error = res.body_reader()
+            if error then
+                return nil, error
+            end
+            if not chunk then
+                break
+            end
+
             if not body then
```

Comment:
Look like this branch is not executed, as `body` has default value `""`?

Reply:
yeah, my fault